### DISCUSSION TITLE
Upload local bim file to CONNECT iTwin?

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#6215] [Q&A]
Can a locally created bim file be upload into a CONNECT iTwin imodel using iTwin Synchronizer?

**Comment by harshalsc** [comment]
Hi @JohnMajerle , No, bim file cannot be synchronized using iTwin Synchronizer.

**Comment by JohnMajerle** [comment]
Can it be done using the imodeljs API?

**Reply by aruniverse** [reply]
Have you tried [BackendHubAccess.createNewIModel](https://github.com/iTwin/itwinjs-core/blob/master/core/backend/src/BackendHubAccess.ts#L254) ?

**Comment by JohnMajerle** [comment]
Hello Arun,
The method createNewIModel() can be used to create a new empty iModel on the hub.  But what API would I be using to push contents into it?  (schemas, elements, etc.)  Is there example code in itwinjs-core that does that?

**Reply by pmconne** [reply]
The documentation for `CreateNewIModelProps` is woefully inadequate, but you can use the [version0](https://github.com/iTwin/itwinjs-core/blob/72eceeb3ecdc3c65e6fc2c9241cec41d622f5cfa/core/backend/src/BackendHubAccess.ts#L180) field to specify a local .bim file to upload as the initial version.
